creusot/src/ctx.rs

```diff
@@ -386,7 +389,10 @@ impl<'tcx, 'sess> TranslationCtx<'tcx> {
         }
     }
 
-    pub(crate) fn span_attr(&self, span: Span) -> Option<why3::declaration::Attribute> {
+    pub(crate) fn span_attr(&mut self, span: Span) -> Option<why3::declaration::Attribute> {
+        if let Some(span) = self.span_map.encode_span(&self.opts, span) {
+            return Some(span);
+        };
```

Comment:
Yes, this function breaks that rule, but perhaps we can take advantage of this PR to move it?